lib/logstash/inputs/decoder_impl.rb

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+require 'java'
+
+java_import org.logstash.tcp.InputLoop
+java_import org.logstash.tcp.Decoder
+
+class DecoderImpl
+
+  include Decoder
+
+  def initialize(codec, tcp)
+    @tcp = tcp
+    @codec = codec
+    @codecs = {}
+    @first_read = {}
+    @addresses = {}
+    @ports = {}
+    @proxy_addresses = {}
+    @proxy_ports = {}
+  end
+
+  def decode(channel_addr, data)
+    if @codecs[channel_addr].nil?
+      @codecs[channel_addr] = @codec.clone
+      @first_read[channel_addr] = true
+    end
+    bytes = Java::byte[data.readableBytes].new
+    data.getBytes(0, bytes)
+    tbuf = String.from_java_bytes bytes, "ASCII-8BIT"
```

Comment:
Is the ASCII-8BIT needed here ? 

I _think_ this will instruct the String object to treat each character as 8bits, which works without question if the original encoding is ASCII7/8 bit. However if the original encoding was something else AND something down the chain tries to do something with the String object (like print it) characters outside the 7/8 bit codec will be (for example printed) incorrect. (another example could be trying split on a unicode character) 

I couldn't find much doc on the exact behavior of JRuby String's implementation, but if it works without the presumption of ASCII-8BIT, I would suggest to remove it.